### ISSUE TITLE
TEST: Remove external file downloads in tests

### ISSUE THREAD

**Issue by asoplata** [jonescompneurolab/hnn-core#936]
Currently, there are several tests that retrieve various files from different external locations on GitHub:

- `hnn_core/tests/test_params.py::test_read_legacy_params` ( https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/tests/test_params.py#L46 ) downloads a file of size 5.7kb to `hnn_core/param/default.param`
    - `hnn_core/tests/test_params.py::TestConvertToJson.test_convert_to_json_legacy()` also downloads this same file
- `hnn_core/tests/test_params.py::test_base_params` ( https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/tests/test_params.py#L66 ) downloads a file of size 6.4kb to `hnn_core/param/base.json`
- `hnn_core/tests/test_params.py::test_remove_nulled_drives` ( https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/tests/test_params.py#L82 ) downloads a file of size 5.7kb to `hnn_core/param/ERPYes100Trials.param`
- `hnn_core/tests/test_dipole.py::test_rmse` ( https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/tests/test_dipole.py#L282 ) downloads a data file of size 5.1kb called `yes_trial_S1_ERP_all_avg.txt` to wherever `pytest` is run from
- `hnn_core/tests/test_parallel_backends.py::test_compare_hnn_core` ( https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/tests/test_parallel_backends.py#L204 ) downloads a data file of size 209kb called `dpl.txt` to wherever `pytest` is run from

These files should all be downloaded and moved into the test assets directory https://github.com/jonescompneurolab/hnn-core/tree/master/hnn_core/tests/assets since they are small, and it is unnecessary to download them every time tests are run (which is frequent!)

**Comment by jasmainak** [comment]
agreed! If it's a small file, definitely add it to a local folder. Remember that git tracks everything, so if you expect the file to change often, it's not a good idea to add to the local repo

**Comment by Freedisch** [comment]
Hi @asoplata, I looked into it. I downloaded these files and updated the testing
